api/go.mod

```diff
@@ -7,20 +7,23 @@ require (
 	github.com/gogo/protobuf v1.3.2
 	github.com/golang/protobuf v1.5.3
 	github.com/grpc-ecosystem/grpc-gateway v1.16.0
```

Comment:
I really like the direction of this PR.

It looks like there's only two remaining references to v1 grpc-gateway... is it possible to switch those to v2 and drop this completely:

```diff
diff --git a/api/etcdserverpb/rpc.pb.go b/api/etcdserverpb/rpc.pb.go
index 9e50ca6b2..4294fc9e5 100644
--- a/api/etcdserverpb/rpc.pb.go
+++ b/api/etcdserverpb/rpc.pb.go
@@ -12,7 +12,7 @@ import (

        _ "github.com/gogo/protobuf/gogoproto"
        proto "github.com/golang/protobuf/proto"
-       _ "github.com/grpc-ecosystem/grpc-gateway/protoc-gen-swagger/options"
+       _ "github.com/grpc-ecosystem/grpc-gateway/v2/protoc-gen-openapiv2/options"
        authpb "go.etcd.io/etcd/api/v3/authpb"
        mvccpb "go.etcd.io/etcd/api/v3/mvccpb"
        _ "go.etcd.io/etcd/api/v3/versionpb"
```

and 

```diff
diff --git a/tests/e2e/v3_curl_test.go b/tests/e2e/v3_curl_test.go
index b994b4897..354e756eb 100644
--- a/tests/e2e/v3_curl_test.go
+++ b/tests/e2e/v3_curl_test.go
@@ -33,7 +33,7 @@ import (
        epb "go.etcd.io/etcd/server/v3/etcdserver/api/v3election/v3electionpb"
        "go.etcd.io/etcd/tests/v3/framework/e2e"

-       "github.com/grpc-ecosystem/grpc-gateway/runtime"
+       "github.com/grpc-ecosystem/grpc-gateway/v2/runtime"
 )

 var apiPrefix = []string{"/v3"}
```

Reply:
I think the new generated code is into https://github.com/etcd-io/etcd/blob/4f3b8bb58d18b46157a3dd9ff3b2f97668621af7/api/etcdserverpb/go.etcd.io/etcd/api/v3/etcdserverpb/rpc.pb.go. Just need to update the genproto.sh and the old deps will be deleted.

Reply:
All grpc-gateway v1 are removed now.